### PR TITLE
feat: Introduce storage catalog

### LINKED ISSUE #177: Switch to catalog api totally.

### PR DESCRIPTION
**Pull request by liurenjie1024**
Close #177 .

In this we introduced `FileSystemCatalog` and delegate all operations to catalog.

### CODE REVIEW

File: icelake/src/catalog/mod.rs [hunk 1]
```diff
@@ -189,9 +265,158 @@ pub enum MetadataUpdate {
...
+    }
+
+    /// Build.
+    pub fn build(self) -> UpdateTable {
+        self.0
+    }
+}
+
+/// Args for creating opendal operator
+pub enum OperatorArgs {
```
Comment (Xuanwo) [inline icelake/src/catalog/mod.rs:324]: Operator args could be very large, how about storing as `HashMap<String, String>` and build `Operator` via `Operator::from_map()`?

Comment (liurenjie1024) [reply icelake/src/catalog/mod.rs:324]: Really cool, I didn't realize this method before.

**Comment by liurenjie1024** [comment]
cc @Xuanwo Any other comments?

File: icelake/examples/read_iceberg_table.rs [hunk 2]
```diff
@@ -1,13 +1,13 @@
 use std::env;
 
 use anyhow::Result;
-use icelake::Table;
+use icelake::catalog::FileSystemCatalog;
```
Comment (Xuanwo) [inline icelake/examples/read_iceberg_table.rs:4]: Do we have better names for `FileSystemCatalog`? File system always refer to POSIX fs or HDFS.

Icelake shoule be able to work on any storage services. How about `StorageCatalog`?

Comment (liurenjie1024) [reply icelake/examples/read_iceberg_table.rs:4]: LGTM

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Pull request merged**